packages/components/src/components/contract-card/contract-card-items/toggle-card-dialog.tsx

```diff
@@ -113,36 +105,36 @@ const ToggleCardDialog = ({
         />
     );
 
-    const toggle_wrapper =
-        should_show_cancellation_warning && isDesktop() ? (
-            <Popover
-                alignment='right'
-                classNameBubble='dc-contract-card-dialog__popover'
-                is_bubble_hover_enabled
-                margin={2}
-                zIndex='2'
-                message={
-                    <PopoverMessageCheckbox
-                        defaultChecked={should_hide_selected}
-                        checkboxLabel={getCardLabels().DONT_SHOW_THIS_AGAIN}
-                        message={getCardLabels().TAKE_PROFIT_LOSS_NOT_AVAILABLE}
-                        name='should_show_cancellation_warning'
-                        onChange={onPopoverCheckboxChange}
-                    />
-                }
-                onBubbleClose={onPopoverClose}
-            >
-                <div className='dc-contract-card-dialog-toggle__wrapper'>{edit_icon}</div>
-            </Popover>
-        ) : (
-            <div className='dc-contract-card-dialog-toggle__wrapper'>{edit_icon}</div>
-        );
-
     return (
         <div onClick={handleClick}>
-            <div ref={toggle_ref} className='dc-contract-card-dialog-toggle' onClick={toggleDialogWrapper}>
-                {is_valid_to_cancel ? toggle_wrapper : edit_icon}
-            </div>
+            {is_valid_to_cancel && should_show_cancellation_warning && isDesktop() ? (
+                <Popover
+                    alignment='right'
+                    classNameBubble='dc-contract-card-dialog__popover-bubble'
+                    className='dc-contract-card-dialog__popover'
+                    is_bubble_hover_enabled
+                    margin={2}
+                    zIndex='2'
+                    message={getCardLabels().TAKE_PROFIT_LOSS_NOT_AVAILABLE}
+                    onBubbleClose={onPopoverClose}
+                >
+                    <button
+                        ref={toggle_ref as unknown as LegacyRef<HTMLButtonElement>}
+                        className='dc-contract-card-dialog-toggle'
+                        onClick={toggleDialogWrapper}
+                    >
+                        {edit_icon}
+                    </button>
+                </Popover>
+            ) : (
+                <button
+                    ref={toggle_ref as unknown as LegacyRef<HTMLButtonElement>}
```

Comment:
Not sure, but will `React.RefObject<HTMLButtonElement>` work?

Reply:
I changed the type because button only accept LegacyRef and typescript throws error

Reply:
Yep, it should work. I will change it